File: app/src/main/java/com/sdex/activityrunner/preferences/SortingPreferences.java
package com.sdex.activityrunner.preferences;

import android.content.SharedPreferences;

import com.sdex.activityrunner.db.application.ApplicationModel;

public class SortingPreferences {

  private final SharedPreferences sharedPreferences;

  public SortingPreferences(SharedPreferences sharedPreferences) {
    this.sharedPreferences = sharedPreferences;
  }

  public String getSortBy() {
    String sortBy = sharedPreferences.getString(SettingsActivity.KEY_SORT_BY,
      SettingsActivity.KEY_SORT_BY_DEFAULT);
    int position = Integer.parseInt(sortBy);
    if (position == 0) {
      return ApplicationModel.NAME;
    } else if (position == 1) {
      return ApplicationModel.PACKAGE_NAME;
    }
    throw new IllegalStateException("Unknown sort by position " + position);
  }

  public String getSortOrder() {
    String sortOrder = sharedPreferences.getString(SettingsActivity.KEY_SORT_ORDER,
      SettingsActivity.KEY_SORT_ORDER_DEFAULT);
    int position = Integer.parseInt(sortOrder);
    if (position == 0) {
      return "ASC";
    } else if (position == 1) {
      return "DESC";
    }
    throw new IllegalStateException("Unknown sort order position " + position);
  }

  public String getSortCaseSensitive() {
    boolean caseSensitive = sharedPreferences.getBoolean(SettingsActivity.KEY_SORT_CASE_SENSITIVE,
      SettingsActivity.KEY_SORT_CASE_SENSITIVE_DEFAULT);
    if (!caseSensitive) {
      return "COLLATE NOCASE";
    }
    return "";
  }
}

File: app/src/main/java/com/sdex/activityrunner/preferences/SortingPreferences.kt
package com.sdex.activityrunner.preferences

import android.content.SharedPreferences
import com.sdex.activityrunner.db.cache.ApplicationModel

class SortingPreferences(private val sharedPreferences: SharedPreferences) {

  val sortBy: String
    get() {
      val sortBy = sharedPreferences.getString(KEY_SORT_BY, KEY_SORT_BY_DEFAULT)
      val position = Integer.parseInt(sortBy ?: KEY_SORT_BY_DEFAULT)
      if (position == 0) {
        return ApplicationModel.NAME
      } else if (position == 1) {
        return ApplicationModel.PACKAGE_NAME
      }
      throw IllegalStateException("Unknown sort by position $position")
    }

  val sortOrder: String
    get() {
      val sortOrder = sharedPreferences.getString(KEY_SORT_ORDER, KEY_SORT_ORDER_DEFAULT)
      val position = Integer.parseInt(sortOrder ?: KEY_SORT_ORDER_DEFAULT)
      if (position == 0) {
        return "ASC"
      } else if (position == 1) {
        return "DESC"
      }
      throw IllegalStateException("Unknown sort order position $position")
    }

  val sortCaseSensitive: String
    get() {
      val caseSensitive = sharedPreferences.getBoolean(KEY_SORT_CASE_SENSITIVE,
        KEY_SORT_CASE_SENSITIVE_DEFAULT)
      return if (!caseSensitive) {
        "COLLATE NOCASE"
      } else ""
    }

  companion object {

    const val KEY_SORT_ORDER = "sort_order"
    const val KEY_SORT_ORDER_DEFAULT = "0"
    const val KEY_SORT_BY = "sort_by"
    const val KEY_SORT_BY_DEFAULT = "0"
    const val KEY_SORT_CASE_SENSITIVE = "sort_case_sensitive"
    const val KEY_SORT_CASE_SENSITIVE_DEFAULT = false
  }
}
